File: java/src/test/java/com/peshchuk/topjslibs/GoogleSearchResultTest.java
package com.peshchuk.topjslibs;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Arrays;
import org.junit.jupiter.api.Test;

public class GoogleSearchResultTest {

  @Test
  public void getLinks() {
    // Arrange
    final var expected = Arrays.asList(
        "https://www.test.de/",
        "https://www.test.de/shop/test-hefte/",
        "https://www.test.de/thema/",
        "https://www.oekotest.de/",
        "https://de.wikipedia.org/wiki/Test_(Zeitschrift)"
    );

    // Act
    final var actual = new GoogleSearchResult(5, () -> Util.getHtmlFrom("google.com.html"))
        .getLinks();

    // Assert
    assertEquals(expected, actual);
  }
}

File: scala/src/test/scala/com/peshchuk/topjslibs/GoogleSearchResultTest.scala
package com.peshchuk.topjslibs

import org.junit.runner.RunWith
import org.scalatest.funsuite.AnyFunSuite
import org.scalatestplus.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class GoogleSearchResultTest extends AnyFunSuite {

  test("should get the first 5 links from the Google search result") {
    // Arrange
    val expected = List(
      "https://www.test.de/",
      "https://www.test.de/shop/test-hefte/",
      "https://www.test.de/thema/",
      "https://www.oekotest.de/",
      "https://de.wikipedia.org/wiki/Test_(Zeitschrift)"
    )
    val googleSearchResult = new GoogleSearchResult(linksLimit = 5,
      fetchHtml = () => Util.getHtmlFrom("google.com.html"))

    // Act
    val actual = googleSearchResult.getLinks

    // Assert
    assert(expected == actual)
  }
}
